code_table.cpp

```diff
@@ -0,0 +1,26 @@
+#include "code_table.h"
+#include <climits>
+#include <stdexcept>
+
+using namespace std;
+
+CodeTable::CodeTable() {
+    for (unsigned int c = 0; c <= UCHAR_MAX; ++c) {
```

Comment:
Maybe instead of using an `unsigned int` variable and converting it to char each time we could iterate with an `unsigned char` variable? like in `StringTable`.